src/cvodes/cvodes_ls.c

```diff
@@ -1850,20 +1869,22 @@ int cvLsSolve(CVodeMem cv_mem, N_Vector b, N_Vector weight,
   nli_inc = 0;
   if (cvls_mem->iterative) {
 #if SUNDIALS_LOGGING_LEVEL >= SUNDIALS_LOGGING_DEBUG
-    if (cvls_mem->LS->ops->resnorm)
-      resnorm = SUNLinSolResNorm(cvls_mem->LS);
+    if (cvls_mem->LS->ops->resnorm) {
+      resnorm = SUNCheckCallLastErrNoRet(SUNLinSolResNorm(cvls_mem->LS));
+    }
 #endif
-    if (cvls_mem->LS->ops->numiters)
-      nli_inc = SUNLinSolNumIters(cvls_mem->LS);
+    if (cvls_mem->LS->ops->numiters) {
+      nli_inc = SUNCheckCallLastErrNoRet(SUNLinSolNumIters(cvls_mem->LS));
+    }
   }
 
 
   /* Increment counters nli and ncfl */
   cvls_mem->nli += nli_inc;
-  if (retval != SUNLS_SUCCESS) cvls_mem->ncfl++;
+  if (ls_status != SUNLS_SUCCESS) cvls_mem->ncfl++;
 
   /* Interpret solver return value  */
-  cvls_mem->last_flag = retval;
+  cvls_mem->last_flag = ls_status;
```

Comment:
Currently, `cvls_mem->last_flag` has type `int`, but `ls_status` has type `SUNLsStatus`.

Given the level of inconsistency in use of `SUNLsStatus`, and the fact that this is just an `int`, I wonder whether we should remove `SUNLsStatus` entirely (similarly for `SUNNlsStatus`).